core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java

```diff
@@ -835,15 +840,21 @@ public void testDefaultWaitForActiveShardsUsesIndexSetting() throws Exception {
         assertEquals(ActiveShardCount.from(requestWaitForActiveShards), request.waitForActiveShards());
     }
 
-    /** test that a primary request is rejected if it arrives at a shard with a wrong allocation id */
-    public void testPrimaryActionRejectsWrongAid() throws Exception {
+    /** test that a primary request is rejected if it arrives at a shard with a wrong allocation id or term */
+    public void testPrimaryActionRejectsWrongAidOrWrongTerm() throws Exception {
         final String index = "test";
         final ShardId shardId = new ShardId(index, "_na_", 0);
         setState(clusterService, state(index, true, ShardRoutingState.STARTED));
+        final ShardRouting primary = clusterService.state().routingTable().shardRoutingTable(shardId).primaryShard();
+        final long primaryTerm = clusterService.state().metaData().index(shardId.getIndexName()).primaryTerm(shardId.id());
         PlainActionFuture<TestResponse> listener = new PlainActionFuture<>();
+        final boolean wrongAllocationId = randomBoolean();
+        final long wrongTerm = primaryTerm + randomIntBetween(1, 10);
```

Comment:
nit: this doesn't allow for the right term and the wrong aid. Something that can't really happen, but is good to test imo.